app/src/androidTest/java/ch/epfl/cs311/wanderwave/model/ProfileConnectionTest.kt

```diff
@@ -2,30 +2,47 @@ package ch.epfl.cs311.wanderwave.model
 
 import android.net.Uri
 import ch.epfl.cs311.wanderwave.model.data.Profile
+import ch.epfl.cs311.wanderwave.model.data.Track
 import ch.epfl.cs311.wanderwave.model.remote.ProfileConnection
+import ch.epfl.cs311.wanderwave.model.remote.TrackConnection
+import com.google.android.gms.tasks.OnSuccessListener
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.DocumentReference
 import com.google.firebase.firestore.DocumentSnapshot
 import com.google.firebase.firestore.FirebaseFirestore
-import com.google.firebase.firestore.Query
 import com.google.firebase.firestore.QuerySnapshot
 import io.mockk.MockKAnnotations
+import io.mockk.coVerify
 import io.mockk.every
 import io.mockk.impl.annotations.RelaxedMockK
 import io.mockk.junit4.MockKRule
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
 import junit.framework.TestCase.assertEquals
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 
 /*
-public constructor Profile(
-    val firstName: String,
-    val lastName: String,
-    val description: String,
-    val numberOfLikes: Int,
-    val isPublic: Boolean,
-    val profilePictureUri: Uri? = null,
-    val spotifyUid: String,
-    val firebaseUid: String
+data class Profile(
+    var firstName: String,
+    var lastName: String,
+    var description: String,
+    var numberOfLikes: Int,
+    var isPublic: Boolean,
+    var profilePictureUri: Uri? = null,
+    var spotifyUid: String,
+    var firebaseUid: String,
+    var topSongs: List<Track> = emptyList(),
+    var chosenSongs: List<Track> = emptyList(),
 )
  */
```

Comment:
This is the profile data class, just to get an idea of what are the parameters of the profile, it has been here since the begining, and I don't think it's a bad thing. But I can delete it.